### PR TITLE
Add io.github.babaissarkar.ssplot

### PR DESCRIPTION
**Pull request by babaissarkar**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
  A simple plotting utility and numerical simulator, it can plot numerical datafiles (format similar to gnuplot) as well as functions and differential/difference equations, suitable for studying dynamical systems.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

File: io.github.babaissarkar.ssplot.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+    "app-id": "io.github.babaissarkar.ssplot",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "23.08",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": "org.freedesktop.Sdk.Extension.openjdk17",
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:6]:
```
(flatpak-builder:2): Json-WARNING **: 12:16:11.496: Failed to deserialize "sdk-extensions" property of type "GStrv" for an object of type "BuilderManifest"
Dependency Sdk: org.freedesktop.Sdk 23.08
```

File: io.github.babaissarkar.ssplot.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "command": "ssplot",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=host:rw"
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:12]: why host filesystem permissions?

File: io.github.babaissarkar.ssplot.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  <metadata_license>CC-BY-4.0</metadata_license>
+  <project_license>LGPL-2.1-or-later</project_license>
+  
+  <description>
+    <p>
+      A simple plotting utility and numerical simulator. It can : <em> Plot datafiles (2D and 3D). </em> Can change appearance of plot. <em> Solve and plot ODE and Difference Equations. </em> Plot functions of one and two variables. * Comes with its own JavaScript based Console for simple calculation tasks.
+    </p>
+  </description>
+  
+  <launchable type="desktop-id">io.github.babaissarkar.ssplot.desktop</launchable>
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.metainfo.xml:17]: missing required developer info, `releases`, `screenshots`, `url`.

File: io.github.babaissarkar.ssplot.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            ]
+        },
+        {
+            "name": "SSPlot",
+            "buildsystem": "simple",
+            "builddir": true,
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "https://raw.githubusercontent.com/babaissarkar/ssplot/master/jar/SSPlot.jar",
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:29]: you DO NOT fetch the file from the master branch.

a checksum is required.

File: io.github.babaissarkar.ssplot.json (hunk 2 above)
Comment (babaissarkar) [reply io.github.babaissarkar.ssplot.json:12]: So that it can open and save files.

Comment (babaissarkar) [reply io.github.babaissarkar.ssplot.json:12]: read plot data and write the resultant image back

File: io.github.babaissarkar.ssplot.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    "app-id": "io.github.babaissarkar.ssplot",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "23.08",
+    "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": "org.freedesktop.Sdk.Extension.openjdk17",
+    "command": "ssplot",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:11]: does this really support wayland? experience show that Java doesn't.

File: io.github.babaissarkar.ssplot.json (hunk 2 above)
Comment (hfiguiere) [reply io.github.babaissarkar.ssplot.json:12]: this doesn't justify `host` at all.

File: io.github.babaissarkar.ssplot.json (hunk 5 above)
Comment (babaissarkar) [reply io.github.babaissarkar.ssplot.json:11]: Okay, will remove that.

File: io.github.babaissarkar.ssplot.json (hunk 2 above)
Comment (babaissarkar) [reply io.github.babaissarkar.ssplot.json:12]: is `--filesystem=home:rw` okay?

File: io.github.babaissarkar.ssplot.desktop
Comment (bbhtt) [inline io.github.babaissarkar.ssplot.desktop:1]: Desktop file should be upstream

File: io.github.babaissarkar.ssplot.metainfo.xml
Comment (bbhtt) [inline io.github.babaissarkar.ssplot.metainfo.xml:1]: metainfo file should be upstream

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

File: io.github.babaissarkar.ssplot.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                    "type": "file",
+                    "url": "https://raw.githubusercontent.com/babaissarkar/ssplot/v2.2.2/SSPlot.AppDir/ssplot.desktop",
+                    "sha256": "bf300c2d694417732053f5930ef8eb3063630c4abcd8afb3d5a5129f07c66901",
+                    "dest-filename": "io.github.babaissarkar.ssplot.desktop"
+                },
+                {
+                    "type": "file",
+                    "url": "https://raw.githubusercontent.com/babaissarkar/ssplot/v2.2.2/io.github.babaissarkar.ssplot.metainfo.xml",
+                    "sha256": "8eb33e09286f2e271b8b2ba98f0ea1f32d0ebcc81b24ebd2e1b09b25e7054300"
+                },
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:34]: these two files are in the tarball. no reason to list them here.

File: io.github.babaissarkar.ssplot.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            "name": "SSPlot",
+            "buildsystem": "simple",
+            "builddir": true,
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/babaissarkar/ssplot/archive/refs/tags/v2.2.2.tar.gz",
+                    "sha256": "3582d04d8d8be2b71ef5104a0d888f505e54b7df3c155647babd3ab6aabde247",
+                    "dest-filename": "ssplot.tar.gz"
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:22]:
```suggestion
                    "sha256": "3582d04d8d8be2b71ef5104a0d888f505e54b7df3c155647babd3ab6aabde247"
```

File: io.github.babaissarkar.ssplot.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            ],
+            
+            "build-commands": [
+                 "sed -i 's/Icon=ssplot/Icon=io.github.babaissarkar.ssplot/' io.github.babaissarkar.ssplot.desktop",
+                 "cp -vr SSPlot.AppDir/usr/bin /app",
+                 "cp -vr SSPlot.AppDir/usr/lib /app",
+                 "cp -vr SSPlot.AppDir/usr/share /app",
+                 "install -Dm644 SSPlot.AppDir/ssplot.png -t $FLATPAK_DEST/share/icons/hicolor/128x128/apps/io.github.babaissarkar.ssplot.png",
+                 "install -Dm644 io.github.babaissarkar.ssplot.metainfo.xml -t $FLATPAK_DEST/share/metainfo",
+                 "install -Dm644 io.github.babaissarkar.ssplot.desktop -t $FLATPAK_DEST/share/applications",
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:48]:
```suggestion
                 "install -Dm644 SSPlot.AppDir/ssplot.desktop -t $FLATPAK_DEST/share/applications/$FLATPAK_ID",
```

File: io.github.babaissarkar.ssplot.json [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                    "sha256": "8eb33e09286f2e271b8b2ba98f0ea1f32d0ebcc81b24ebd2e1b09b25e7054300"
+                },
+                {
+                    "type": "file",
+                    "path": "ssplot-run"
+                }
+            ],
+            
+            "build-commands": [
+                 "sed -i 's/Icon=ssplot/Icon=io.github.babaissarkar.ssplot/' io.github.babaissarkar.ssplot.desktop",
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:42]: never use `sed`, it has no failure mode
```suggestion
                 "desktop-file-edit --set-icon=$FLATPAK_ID SSPlot.AppDir/ssplot.desktop",
```

File: io.github.babaissarkar.ssplot.json [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+                 "cp -vr SSPlot.AppDir/usr/bin /app",
```
Comment (hfiguiere) [inline io.github.babaissarkar.ssplot.json:43]: how will this run on aarch64 ?

Comment (babaissarkar) [reply io.github.babaissarkar.ssplot.json:43]: It's the `cp` command, right?

Comment (hfiguiere) [reply io.github.babaissarkar.ssplot.json:43]: I'm talking about copy x86_64 binaries in a aarch64 flatpak. While Java is compile once run everywhere, the JRE definitely is not.

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

**Comment by babaissarkar** [comment]
What's causing this :
Errors were raised during this compose run:
```
io.github.babaissarkar.ssplot
  E: gui-app-without-icon
general
  E: filters-but-no-output
```
I did add a `<icon/>` tag in the metainfo file.

**Comment by bbhtt** [comment]
Desktop applications don't need an icon tag what you need is https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#launchable, make sure the desktop filename matches ID and icon name matches ID and `Icon` key matches icon name

File: io.github.babaissarkar.ssplot.json [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+                {
+                    "type": "file",
+                    "path": "ssplot-run"
+                }
+            ],
+            "build-commands": [
+                 "desktop-file-edit --set-icon=$FLATPAK_ID SSPlot.AppDir/ssplot.desktop",
+                 "install -Dm644 SSPlot.AppDir/ssplot.png -t $FLATPAK_DEST/share/icons/hicolor/128x128/apps/io.github.babaissarkar.ssplot.png",
+                 "install -Dm644 $FLATPAK_ID.metainfo.xml -t $FLATPAK_DEST/share/metainfo",
+                 "install -Dm644 SSPlot.AppDir/ssplot.desktop -t $FLATPAK_DEST/share/applications/$FLATPAK_ID",
```
Comment (bbhtt) [inline io.github.babaissarkar.ssplot.json:38]:
```suggestion
                 "install -Dm644 SSPlot.AppDir/ssplot.desktop $FLATPAK_DEST/share/applications/${FLATPAK_ID}.desktop",
```

File: io.github.babaissarkar.ssplot.json [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+                    "sha256": "3582d04d8d8be2b71ef5104a0d888f505e54b7df3c155647babd3ab6aabde247"
+                },
+                {
+                    "type": "file",
+                    "path": "ssplot-run"
+                }
+            ],
+            "build-commands": [
+                 "desktop-file-edit --set-icon=$FLATPAK_ID SSPlot.AppDir/ssplot.desktop",
+                 "install -Dm644 SSPlot.AppDir/ssplot.png -t $FLATPAK_DEST/share/icons/hicolor/128x128/apps/io.github.babaissarkar.ssplot.png",
```
Comment (bbhtt) [inline io.github.babaissarkar.ssplot.json:36]:
```suggestion
                 "install -Dm644 SSPlot.AppDir/ssplot.png $FLATPAK_DEST/share/icons/hicolor/128x128/apps/$FLATPAK_ID.png",
```

Comment (bbhtt) [reply io.github.babaissarkar.ssplot.json:36]: -t expects the target directory not the filname

File: io.github.babaissarkar.ssplot.json (hunk 11 above)
Comment (hfiguiere) [reply io.github.babaissarkar.ssplot.json:38]: sorry I messed up that one. This suggestion is correct.

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

File: io.github.babaissarkar.ssplot.json (hunk 2 above)
Comment (bbhtt) [reply io.github.babaissarkar.ssplot.json]: changed to xdg-documents

**Comment by bbhtt** [comment]
```
 flatpak run io.github.babaissarkar.ssplot
Error: Unable to access jarfile /app/share/ssplot/ssplot.jar

```

**Comment by babaissarkar** [comment]
bot, build io.github.babaissarkar.ssplot

**Pull request closed without merging**